clang/test/Driver/sycl-offload.cpp

```diff
@@ -47,3 +47,60 @@
 // CHECK_S_LLVM: clang{{.*}} "-fsycl-is-device"{{.*}} "-emit-llvm"{{.*}} "-o" "[[DEVICE:.+\.ll]]"
 // CHECK_S_LLVM: clang{{.*}} "-fsycl-is-host"{{.*}} "-emit-llvm"{{.*}} "-o" "[[HOST:.+\.ll]]"
 // CHECK_S_LLVM: clang-offload-bundler{{.*}} "-type=ll"{{.*}} "-inputs=[[DEVICE]],[[HOST]]"
+
+/// Check for default device triple compilations based on object, archive or
+/// forced from command line.
+// RUN:  echo "void foo();" > %t_dummy.cpp
+// RUN:  %clang -fsycl -c %t_dummy.cpp -o %t_dummy.o
+// RUN:  llvm-ar cr %t_dummy.a %t_dummy.o
+// RUN:  touch %t_empty.o
+// RUN:  %clangxx -### -fsycl -fsycl-targets=spir64_x86_64 %t_dummy.o %s 2>&1 \
+// RUN:    | FileCheck -check-prefix IMPLIED_DEVICE_OBJ %s
+// RUN:  %clangxx -### -fsycl -fsycl-targets=spir64_fpga %t_dummy.o %s 2>&1 \
+// RUN:    | FileCheck -check-prefix IMPLIED_DEVICE_OBJ %s
+// RUN:  %clangxx -### -fsycl -fsycl-targets=spir64_gen %t_dummy.o %s 2>&1 \
+// RUN:    | FileCheck -check-prefix IMPLIED_DEVICE_OBJ %s
+// RUN:  %clangxx -### -fsycl -fintelfpga %t_dummy.o %s 2>&1 \
+// RUN:    | FileCheck -check-prefix IMPLIED_DEVICE_OBJ %s
+// RUN:  %clangxx -### -fsycl -fsycl-add-spirv -fsycl-targets=spir64_x86_64 %t_empty.o %s 2>&1 \
+// RUN:    | FileCheck -check-prefix IMPLIED_DEVICE_OBJ %s
+// RUN:  %clangxx -### -fsycl -fsycl-add-spirv -fsycl-targets=spir64_fpga %t_empty.o %s 2>&1 \
+// RUN:    | FileCheck -check-prefix IMPLIED_DEVICE_OBJ %s
+// RUN:  %clangxx -### -fsycl -fsycl-add-spirv -fsycl-targets=spir64_gen %t_empty.o %s 2>&1 \
+// RUN:    | FileCheck -check-prefix IMPLIED_DEVICE_OBJ %s
+// RUN:  %clangxx -### -fsycl -fsycl-add-spirv -fintelfpga %t_empty.o %s 2>&1 \
+// RUN:    | FileCheck -check-prefix IMPLIED_DEVICE_OBJ %s
```

Comment:
Looking back at this now, I'm not sure if I understand the purpose of `-fsycl-add-spirv`. Granted, it doesn't hurt at all, but the use-cases I initially thought relevant no longer appear to be. Are there cases where a simple `-fsycl-targets` extension with the default triple doesn't do a good job?

Were the option to be retained, it could make sense to extend its effect for the source inputs as well (or is it working already?)

Reply:
Functionality of `-fsycl-add-spirv` is just the equivalent of adding `spir64` to `-fsycl-targets`.  I'm OK with not adding it.